examples/flight_udtf_llm_example.py

```diff
@@ -0,0 +1,82 @@
+import json
+import os
+from pathlib import Path
+from urllib.parse import unquote_plus
+
+import pandas as pd
+import requests
+
+import xorq as xo
+
+
+request_timeout = 3
+
+
+def extract_sentiment(text):
+    url = "https://api.openai.com/v1/chat/completions"
+    headers = {
+        "Authorization": f"Bearer {os.environ['OPENAI_API_KEY']}",
+        "Content-Type": "application/json",
+    }
+    data = {
+        "model": "gpt-3.5-turbo",
+        "messages": [
+            {
+                "role": "system",
+                "content": "You are an AI language model trained to analyze and detect the sentiment of hackernews forum comments.",
+            },
+            {
+                "role": "user",
+                "content": f"Analyze the following hackernews comment and determine if the sentiment is: positive, negative or neutral. Return only a single word, either POSITIVE, NEGATIVE or NEUTRAL: {text}",
+            },
+        ],
+    }
+
+    response = requests.post(url, headers=headers, data=json.dumps(data))
+
+    if response.status_code == 200:
+        result = response.json()
+        sentiment = result["choices"][0]["message"]["content"].strip().lower()
+        return sentiment
+    else:
+        return f"Error: {response.status_code}, {response.text}"
+
+
+def get_hackernews_sentiment_batch(df: pd.DataFrame):
+    df = df.assign(text=df["text"].apply(lambda x: unquote_plus(x.decode("utf-8"))))
+    return df.assign(sentiment=df.text.apply(extract_sentiment))
+
+
+schema_in = xo.schema(
+    {
+        "text": "!binary",
+    }
+)
+schema_out = xo.schema({"text": "!str", "sentiment": "!str"})
+
+do_hackernews_sentiment_udxf = xo.expr.relations.flight_udxf(
+    process_df=get_hackernews_sentiment_batch,
+    maybe_schema_in=schema_in.to_pyarrow(),
+    maybe_schema_out=schema_out.to_pyarrow(),
+    name="HackerNewsSentimentAnalyzer",
+    # make_server=lambda: FlightServer(flight_url=FlightUrl(port=5005)),
+)
+
+path = Path(__file__).parent / "data" / "hn_nano.parquet"
```

Comment:
Not sure if we should be checking in data for examples. If this is required, it should be something like pins.